src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,789 @@
+            achAccount.setAchTransactionType(getDirectDepositTransactionType());
+            achAccount.setBankRoutingNumber(achDetail.getBankRoutingNumber());
+            achAccount.setBankAccountNumber(achDetail.getBankAccountNumber());
+            achAccount.setBankAccountTypeCode(getACHTransactionCode(achDetail.getBankAccountType()));
+            if (StringUtils.isNotBlank(payee.getNameUnmasked())) {
+                achAccount.setPayeeName(payee.getNameUnmasked());
+            }
+            if (StringUtils.isNotBlank(payee.getEmailAddressUnmasked())) {
+                achAccount.setPayeeEmailAddress(payee.getEmailAddressUnmasked());
+            }
+            achAccount.setActive(true);
+
+            // Add a note indicating that this document was generated by a batch process.
+            addNote(paatDocument, parameterService.getParameterValueAsString(
+                    PayeeACHAccountExtractStep.class, CUPdpParameterConstants.GENERATED_PAYEE_ACH_ACCOUNT_DOC_NOTE_TEXT));
+
+            // Route the document and send notifications.
+            paatDocument = (MaintenanceDocument) documentService.routeDocument(paatDocument, KFSConstants.EMPTY_STRING, null);
+            sendPayeeACHAccountAddOrUpdateEmail((PayeeACHAccount) paatDocument.getNewMaintainableObject().getDataObject(), payee,
+                    getPayeeACHAccountAddOrUpdateEmailSubject(true), getUnresolvedPayeeACHAccountAddOrUpdateEmailBody(true));
+
+            return processingError;
+            
+        } catch (Exception e) {
+            LOG.error("addACHAccount STE " + e.getStackTrace() + e.toString());
+            LOG.error(getFailRequestMessage(e));
+            processingError = new String("addACHAccount: " + achDetail.getLogData() + " STE was generated. " + getFailRequestMessage(e));
+            return processingError;
+        }
+    }
+
+    /**
+     * Checks if the existing ACH account matches the data fed in from the input file, and creates and routes
+     * a new PAAT document to edit the account if necessary.
+     * 
+     * NOTE: The current implementation just logs match/mismatch status without making updates.
+     * This method will indicate that valid processing occurred whenever it is caled until it 
+     * is modified in the future to perform actual updates.
+     */
+    protected String updateACHAccountIfNecessary(Person payee, PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
+        StringBuilder processingResults = new StringBuilder();
+        processingResults.append("Update functionality NOT implemented yet. Payee \"")
+        .append(achDetail.getNetID()).append("\" data being checked for possible mismatch. ");
+
+        if (!StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())
+                || !StringUtils.equals(achDetail.getBankAccountNumber(), achAccount.getBankAccountNumber())) {
+            /*

```

Comment:
Is there a JIRA for this "future enhancement"? In general, I prefer to have things like this in the issue tracking system vs. a comment in the code.